framework/src/meshgenerators/XYDelaunayGenerator.C

```diff
@@ -409,6 +411,11 @@ XYDelaunayGenerator::generate()
       }
       mooseAssert(found_inner_sides == np, "Failed to find full boundary around meshed hole");
 
+      // Retrieve subdomain name map from the mesh to be stitched and insert it into the main
+      // subdomain map
+      const auto & increment_subdomain_map = hole_mesh.get_subdomain_name_map();
+      main_subdomain_map.insert(increment_subdomain_map.begin(), increment_subdomain_map.end());
```

Comment:
Is this really where we want to fix an issue with stitching?  If `libMesh::two_plus_two()` is returning 5, surely we want to fix `two_plus_two()`, not just subtract 1 after every time we call it.

There's some fairly tricky code in `stitch_meshes` now, to allow us to merge subdomains by name while remapping ids as necessary; I haven't enabled it in MOOSE because there's a couple tests that are breaking with it, but it looks like this change would be completely incompatible with it.

Reply:
It's debatable (or tricky) whether we want stitching to always merge (based on names or ids) or to remap.
If we still have all 3 options with your patch we should be good here?